### PR TITLE
latch exec

### PR DESCRIPTION
**Pull request by ayushkamat**

### CODE REVIEW

File: latch_cli/services/execute/main.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                },
+            })
+        )
+
+
+async def connect(egn_info: EGNNode, container_info: Optional[ContainerNode]):
+    loop = asyncio.get_event_loop()
+
+    resize_event_queue: asyncio.Queue[tuple[int, int]] = asyncio.Queue()
+    await resize_event_queue.put(os.get_terminal_size())
```
Comment (maximsmol) [inline latch_cli/services/execute/main.py:128]:
```suggestion
    await handle_resize(resize_event_queue)
```

File: latch_cli/services/execute/main.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    loop.add_signal_handler(
+        signal.SIGWINCH,
+        lambda: asyncio.create_task(handle_resize(resize_event_queue)),
+    )
+
+    reader, writer = await get_stdio_streams()
```
Comment (maximsmol) [inline latch_cli/services/execute/main.py:135]:
```suggestion
    local_stdin, local_stdout = await get_stdio_streams()
```

File: latch_cli/services/execute/main.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+    async with websockets.connect(
+        urlparse(urljoin("http://localhost:5000", "/workflows/cli/shell"))
```
Comment (maximsmol) [inline latch_cli/services/execute/main.py:138]: `localhost:5000` is a bad choice
I'd do like `5554` (random number)

`5000` already has a daemon running on mac by default and is likely to conflict with other dev servers

Comment (maximsmol) [reply latch_cli/services/execute/main.py:138]: unless this is just for debugging? in which case this is just probably wrong

File: latch_cli/services/execute/main.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        ._replace(scheme="ws")
+        .geturl(),
+        close_timeout=0,
+        extra_headers={"Authorization": get_auth_header()},
+    ) as ws:
+        request = {
+            "egn_id": f'{egn_info["id"]}',
```
Comment (maximsmol) [inline latch_cli/services/execute/main.py:145]:
```suggestion
            "egn_id": str(egn_info["id"]),
```

**Review by maximsmol — CHANGES_REQUESTED** [review]
smallish

File: latch_cli/services/execute/main.py (hunk 3 above)
Comment (ayushkamat) [reply latch_cli/services/execute/main.py:138]: just for debugging, will remove

File: latch_cli/services/execute/main.py (hunk 4 above)
Comment (ayushkamat) [reply latch_cli/services/execute/main.py:145]: lol i have no idea why i threw this in an fstring in the first place its a str already

**Pull request merged**